File: week2/Ariling/prgs-110 옮기기/main.js
function solution(s) {
  var answer = [];
  for (let i = 0; i < s.length; i++) {
    var stack = [];
    var pattern_cnt = 0;
    for (let j = 0; j < s[i].length; j++) {
      if (s[i][j] === "1") {
        stack.push(s[i][j]);
      } else {
        if (stack.length >= 2 && stack.at(-1) === "1" && stack.at(-2) == "1") {
          stack.pop();
          stack.pop();
          pattern_cnt++;
        } else {
          stack.push(s[i][j]);
        }
      }
    }
    content = stack.join("");
    let found = false;
    for (let k = content.length - 1; k >= 0; k--) {
      if (content[k] === "0") {
        answer.push(
          content.slice(0, k + 1) +
            "110".repeat(pattern_cnt) +
            content.slice(k + 1, content.length)
        );
        found = true;
        break;
      }
    }
    if (!found) {
      answer.push("110".repeat(pattern_cnt) + content);
    }
  }
  return answer;
}

File: week2/Ariling/prgs-110 옮기기/main.py
def solution(s):
    answer = []
    for x in s:
        # 모든 110 찾기
        # 110 찾으면 변형된 x에서 또 찾기
        cnt = 0
        stack = []
        for x_one in x:
            if x_one == '1':
                stack.append(x_one)
            else:
                if len(stack) >= 2 and stack[-1] == '1' and stack[-2] == '1':
                    stack.pop()
                    stack.pop()
                    cnt += 1
                else:
                    stack.append(x_one)
        x = ''.join(stack)
        # 뒤에서부터 체크해서 0이 나오면 그 0 뒤에 모든 110 붙이기
        # 0 없으면 나머지가 전부 1로 구성되어 있으므로 모든 110을 앞에 붙이기
        for i in range(len(x) - 1, -1, -1):
            if x[i] == '0':
                answer.append(x[:i + 1] + ('110' * cnt) + x[i + 1:])
                break
        else:
            answer.append('110' * cnt + x)

    return answer